pkg/resourcecreator/deployment.go

```diff
@@ -42,12 +42,25 @@ func deploymentSpec(app *nais.Application, opts ResourceOptions) (*appsv1.Deploy
 	if err != nil {
 		return nil, err
 	}
-	return &appsv1.DeploymentSpec{
+	deploymentSpec := &appsv1.DeploymentSpec{
```

Comment:
Nitpicking; consider instantiating the `Strategy` object beforehand, so that mutation of the `deploymentSpec` variable isn't needed.